docs/content/4.sdk/2.getting-started/2.middleware-module.md

```diff
@@ -0,0 +1,471 @@
+};
+
+const sdk = initSDK({
+  commerce: buildModule(middlewareModule<Endpoints>, {
+    apiUrl: "http://localhost:8181/commerce",
+  }),
+});
+
+const product = await sdk.commerce.getProduct({ id: "123" });
+```
+
+In Storefront Starter, you can find the `UnifiedEndpoints` type in `storefront-middleware/types.ts` file. It's a type definition for the endpoints used in the middleware.
+
+```ts
+import { UnifiedEndpoints } from "storefront-middleware/types";
+
+export const sdk = initSDK({
+  commerce: buildModule(middlewareModule<UnifiedEndpoints>, {
+    apiUrl: "http://localhost:8181/commerce",
+  }),
+});
+
+const product = await sdk.commerce.getProduct({ id: "123" });
+```
+
+Also, you can import the `Endpoints` type from integration packages.
+
+```ts
+import { Endpoints } from "@vsf-enterprise/sapcc-api";
+
+export const sdk = initSDK({
+  commerce: buildModule(middlewareModule<Endpoints>, {
+    apiUrl: "http://localhost:8181/commerce",
+  }),
+});
+```
+
+### Options
+
+The `middlewareModule` accepts the following options:
+
+- `apiUrl` - the URL of the middleware server,
+- `ssrApiUrl` - (Optional) the URL of the middleware server during SSR,
+- `defaultRequestConfig` - (Optional) default request config for each request,
+- `httpClient` - (Optional) a custom HTTP client,
+- `errorHandler` - (Optional) a custom error handler for HTTP requests.
+
+Example configuration:
+
+```ts
+import { initSDK, buildModule, middlewareModule } from "@vue-storefront/sdk";
+import { UnifiedEndpoints } from "storefront-middleware/types";
+
+export const sdk = initSDK({
+  commerce: buildModule(middlewareModule<UnifiedEndpoints>, {
+    apiUrl: "/api/commerce",
+    ssrApiUrl: "http://localhost:8181/commerce",
+    defaultRequestConfig: {
+      headers: {
+        "X-Api-Key": "123",
+      },
+    },
+    httpClient: async (url, params, config) => {
+      // Custom HTTP client
+    },
+    errorHandler: ({ error, methodName, url, params, config, httpClient }) => {
+      // Custom error handler
+    },
+  }),
+});
+```
+
+## Usage
+
+Once you have added the `middlewareModule` to your SDK, you can use it to make requests to the Server Middleware.
+
+```ts
+const product = await sdk.commerce.getProduct({ id: "123" });
+```
+
+Additionally, each method of this module allows you to pass a custom request configuration. To do it, you need to import `prepareConfig` helper from `@vue-storefront/sdk` package.
+
+```ts
+import { prepareConfig } from "@vue-storefront/sdk";
+
+const product = await sdk.commerce.getProduct(
+  { id: "123" },
+  prepareConfig({
+    method: "GET",
+    headers: {
+      "X-Custom-Header": "123",
+    },
+  })
+);
+```
+
+## Examples
+
+### Send a `GET` request
+
+By default, each request is a `POST` request. You can change it by passing a custom request configuration.
+
+```ts
+const product = await sdk.commerce.getProduct(
+  { id: "123" },
+  prepareConfig({
+    method: "GET",
+  })
```

Comment:
I think we should also update the Caching guide and include the `middlewareModule` there (btw. I remember that we planned to prepare an extension which simplifies setting up the Cache-Control header, do we still want to implement it?)

Reply:
I heard that this guide and need for such extension is no longer necessary as it's going to be done by Michaels CDN approach.